.github/workflows/r.yml

```diff
@@ -244,7 +244,7 @@ jobs:
         config:
         - { rtools: 35, rversion: "3.6" }
         - { rtools: 40, rversion: "4.1" }
-        # TODO: Once R 4.2 comes out we can switch to devel + 4.2
+        - { rtools: 42, rversion: "4.2" }
```

Comment:
This is probably fine (and we should try it and see if it's an issue...) but this _does_ add an additional build to the CI that runs on PRs|commits, and we typically try and keep those as sparse as we can.

But given the differences between roots + windows and our struggles maintaining them, adding a job for not is a good plan — but we might need to reassess if it adds too much waiting

Reply:
Would you mind making a follow up jira for "we shouldn't run lintr on each rtools + windows combo"? _too much_ is very much subjective, and semi-dependent on other apache project's use of runners. Let's merge and see how it is — I just wanted to flag that as a possible thing we need to think about later 🙂

Reply:
Will do. Follow-up Jira was on my to-do list for today. [ARROW-16541](https://issues.apache.org/jira/browse/ARROW-16541)